ENCODE_replaced_cleaner.py

```diff
@@ -0,0 +1,115 @@
+import argparse
+import os.path
+import encodedcc
+import requests
+import urllib.parse
+from time import sleep
+
+import sys
+GET_HEADERS = {'accept': 'application/json'}
+
+
+EPILOG = '''
+For more details:
+
+        %(prog)s --help
+'''
+'''
+Example command:
+python3 ENCODE_replaced_cleaner.py --keyfile keypairs.json --key test
+'''
+
+
+def getArgs():
+    parser = argparse.ArgumentParser(
+        description=__doc__, epilog=EPILOG,
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+    parser.add_argument('--key',
+                        default='default',
+                        help="The keypair identifier from the keyfile.  \
+                        Default is --key=default")
+    parser.add_argument('--keyfile',
+                        default=os.path.expanduser("~/keypairs.json"),
+                        help="The keypair file.  Default is --keyfile=%s" % (os.path.expanduser("~/keypairs.json")))
+    parser.add_argument('--debug',
+                        default=False,
+                        action='store_true',
+                        help="Print debug messages.  Default is False.")
+    args = parser.parse_args()
+    return args
+
+
+def retreive_list_of_replaced(object_to_inspect_acc,
+                              connection):
+    to_return_list = [object_to_inspect_acc]
+    objects_to_inspect = encodedcc.get_ENCODE('search/?type=Item&accession=' +
+                                              object_to_inspect_acc,
+                                              connection)['@graph']
+    if objects_to_inspect:
+        for object_to_inspect in objects_to_inspect:
+            if object_to_inspect.get('alternate_accessions'):
+                for acc in object_to_inspect.get('alternate_accessions'):
+                    to_return_list.extend(
+                        retreive_list_of_replaced(acc, connection))
+                return to_return_list
+            else:
+                return to_return_list
+    else:
+        return to_return_list
+
+
+def main():
+    args = getArgs()
+    key = encodedcc.ENC_Key(args.keyfile, args.key)
+    connection = encodedcc.ENC_Connection(key)
+    profiles = encodedcc.get_ENCODE("/profiles/", connection)
+    for object_type in profiles.keys():
+        uuid_2_alternate_accessions = {}
+        alternate_accessions_sets = []
+        objects = encodedcc.get_ENCODE('search/?type=' + object_type,
+                                       connection)['@graph']
+        for entry in objects:
+            if entry.get('alternate_accessions'):
+                replaced_objects_accessions = []
+                for acc in entry.get('alternate_accessions'):
+                    replaced_objects_accessions.extend(
+                        retreive_list_of_replaced(acc,
+                                                  connection))
+                if sorted(list(set(replaced_objects_accessions))) != sorted(
```

Comment:
The system should never allow the same accession to be present in two different alternate accessions lists, but I am afraid there are ways to go around it. So the list of replaced_objects_accessions should be turned to set (to eliminate potential duplications) and then back into list, that I want to compare to the actual list of accessions in alternate_accessions, I sorted to ensure equality between unsorted lists of same accessions.